internal/p2p/peermanager.go

```diff
@@ -1350,7 +1358,9 @@ type peerInfo struct {
 	Height        int64
 	FixedScore    PeerScore // mainly for tests
 
-	MutableScore int64 // updated by router
+	MutableScore PeerScore // updated by router
+
+	ConsecSuccessfulBlocks int64
```

Comment:
Maybe add another one called NumOfTimeouts?

Reply:
the timeout will trigger the Disconnected() method which has counted to decrease score. the flow is: 

timeout detection -> peer.didTimeout = true -> pool.removePeer(id, true) -> peerManager.Disconnected()